source/common/http/http2/codec_impl.h

```diff
@@ -438,7 +438,11 @@ class ConnectionImpl : public virtual Connection,
     ClientStreamImpl(ConnectionImpl& parent, uint32_t buffer_limit,
                      ResponseDecoder& response_decoder)
         : StreamImpl(parent, buffer_limit), response_decoder_(response_decoder),
-          headers_or_trailers_(ResponseHeaderMapImpl::create()) {}
+          headers_or_trailers_([this]() {
+            auto headers =
+                ResponseHeaderMapImpl::create(parent_.max_headers_kb_, parent_.max_headers_count_);
+            return headers;
+          }()) {}
```

Comment:
here and elsewhere: Would it also work to inline, i.e., headers_or_trailers_(ResponseHeaderMapImpl::create(..., ...))?

Reply:
Yeah, good catch! Done.